### ISSUE TITLE
Tab Groups integration

### ISSUE THREAD

**Issue by darkdh** [brave/browser-laptop#2334]
**Did you search for similar issues before submitting this one?**
Yes

Could we integrate this [awesome extension](https://addons.mozilla.org/en-US/firefox/addon/tab-groups-panorama/) originated from Firefox but was removed in Firefox 45 ?

With this feature, users can easily organized their tabs by intuition and quick switch between groups they managed.

**Comment by bbondy** [comment]
CC @bradleyrichter

**Comment by bradleyrichter** [comment]
@darkdh Thanks for the suggestion. We can build something much better. I think it was dropped for not being used but that may have been due to it being hard to discover.

Originally I wanted our tab sets UI to accommodate this basic functionality. You could organize them by dragging into tab sets as you needed, and we can enable previews on mouse hover of the tab set.

We have some of this slated for 1.1.

Regardless, I would love to hear what part of the FX tab grouping you used most so we can capture the most useful part of the functionality.


thanks - Brad

**Comment by darkdh** [comment]
Thanks @bradleyrichter. I'm looking forward to this function.

**Comment by paanvaannd** [comment]
@bradleyrichter Same! I can't wait to use this function as well. Switched over from Vivaldi to see how Brave was doing and while Brave is definitely snappier, I sorely miss tab stacking.

A further suggestion for stacking: as Vivaldi does, I would recommend an option to group all tabs from a similar host (e.x. all tabs from GitHub) together through a hotkey or right-click option.

**Comment by lucidNTR** [comment]
firefox tab groups had the problem of beeing organized in 2d so they broke all the time when the window was resized and the second issue was they were really really slow.

vivaldis tab groups are more like the tab pages in brave but with inverted hirarchy, adding this in addition to tab pages seems too much brain needed to keep track.

is threre a place to discuss mockup screens before we start implementing? or should we just uplaod screenshots  to this thread?

**Comment by lucidNTR** [comment]
slept about this. i think we should use the current really great tab pages ui to implement tab groups and if there are too many tabs in one group to fit one row, just scroll through them as in firefox. if we have to support tab groups and tab pages this will be too confusing, especially as like someone mentioned before the tabs would swap between pages depending on where tabs are closed so its hard to find them.

**Comment by neomodern** [comment]
Use the format of [Tab Groups Helper](https://addons.mozilla.org/en-US/firefox/addon/tab-groups-helper/?src=search):
![Tab Groups Helper](http://i.imgur.com/NR3AJev.png)

Top row being the tab group itself, the bottom row being all the tabs in the current open tab group.

Overview of all tab groups:
![Tab Groups Helper](http://i.imgur.com/QMujYPN.png)

Very easy to select multiple for deletion or drag-and-drop between tab groups.

Also I only came across this earlier today but [Tabs Outliner](https://chrome.google.com/webstore/detail/tabs-outliner/eggkanocgddhmamlbiijnphhppkpkmkl) for Chrome also looks genius for this process and may be even better for an overview of all tab groups.

Related: [The Great Suspender](https://github.com/brave/browser-laptop/issues/4600#issuecomment-253558339) for suspending inactive tabs.

**Comment by simonwiles** [comment]
[Tab Groups](https://addons.mozilla.org/en-GB/firefox/addon/tab-groups-panorama/) is absolutely awesome, and I'll be looking seriously for a new browser when Firefox 57 comes around and it stops working.  What's most useful is the ability to dismiss and recall (often large) groups of tabs.  It's sooo much better than bookmarking.

**Comment by eljuno** [comment]
+1 from community https://community.brave.com/t/tab-groups-extension-request/6273?u=eljuno

**Comment by paanvaannd** [comment]
@simonwiles Have you looked into Waterfox? I was looking for alternatives recently and just found out about it. The project is keeping support for legacy Firefox extensions so they should all work on there unless one messes with the e10s settings, but e10s is disabled by default to keep legacy extension support running for as long as possible.

**Comment by simonwiles** [comment]
@paanvaannd no, I'd not seen that before -- thanks, it looks promising!

**Comment by justinfiore** [comment]
Any update on this issue? It seems like there hasn't been any activity in about 9 months. Tab Groups would be an awesome feature. I was very disappointed when Firefox got rid of it, and if Brave had this feature, it would easily become my primary browser immediately.
I have always been wary of using extensions that perform Tab Groups functionality, because they typically require access to all pages that are viewed (which is bad from a privacy/security standpoint).

So is this feature going to become a reality in Brave?

**Comment by bsclifton** [comment]
@justinfiore we're in process of moving from the current Brave to [brave-core](https://github.com/brave/brave-browser) which will fix a lot of issues we have as well as get us extension support (meaning you could use extensions like [Tabs Outliner](https://chrome.google.com/webstore/detail/tabs-outliner/eggkanocgddhmamlbiijnphhppkpkmkl)

We won't be able to resolve this specific feature request with `browser-laptop` (the current version of the product), so I'm going to close as `wontfix`. However, we do have a similar issue in brave-core which I'd encourage folks to track / +1 / leave a comment on 😄 
https://github.com/brave/brave-browser/issues/373
